pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/vencord.nix

```diff
@@ -0,0 +1,23 @@
+{
+  runCommand,
+  lib,
+  curl,
+  p7zip,
+}:
+# Fetchzip doesn't recognize the crx file as zip.
```

Comment:
```suggestion
# fetchzip doesn't recognize the crx file as zip.
```
please try setting name to something that ends in zip and then it should work

Reply:
I don't have control over the filename from the url since it's from the chrome webstore. I couldn't find a way to change the filename after the fetching and before the extracting. Do you have any idea for what to do about it?

Reply:
You can use the `extension` argument to `fetchzip`. Unfortunately, though, this fails: CRX extensions have a signature attached to them outside of the zip data. `unzip` succeeds, but returns 1 because it causes a warning, and thus `unpackFile` considers it to have failed.

Maybe this is an acceptable, if somewhat hacky, compromise. You can get unpackFile to pick the 7zip hook instead, and that should actually work:
```nix
fetchzip {
    url = "https://clients2.google.com/service/update2/crx?response=redirect&acceptformat=crx2,crx3&prodversion=109.0&x=id%3Dcbghhgpcnddeihccjmnadmkaejncjndb%26installsource%3Dondemand%26uc";
    extension = "7z";
    nativeBuildInputs = [ pkgs.p7zip ];
    stripRoot = false;
    hash = "sha256-JYpS9BD/O8PtXKW8OzAfmtiJeIIjY5E4SeBREWSUz9w=";
}
```

If that is an acceptable compromise though, the hack (using 7zip and .7z even though it's actually a zip archive) should be commented on most likely.

Reply:
Nevermind. I am editing this PR and building Vencord web extension from source instead of fetching it